.github/actions/setup-k8s/action.yaml

```diff
@@ -0,0 +1,51 @@
+---
+name: 'Setup K8S'
+runs:
+  using: 'composite'
+  steps:
+    - name: Install Kubernetes tools
+      uses: ./.github/actions/install-kubernetes-tools
+      with:
+        install-kubectl: true
+        install-k3d: true
+
+    - name: Adjust GitHub worker node
+      uses: ./.github/actions/adjust-github-node
+      with:
+        adjust-host-for-k3d: true
+        mount-docker-data: true
+
+    - name: Run K3D test cluster
+      shell: bash
+      run: |
+        make k3d-cluster-up
+
+    - name: Build Everest API Server binary
+      shell: bash
+      run: |
+        make build-debug
+
+    - name: Build Everest API server docker image
+      shell: bash
+      run: |
+        make docker-build
+
+    - name: Upload Everest API server image to K3D test cluster
+      shell: bash
+      run: |
+        make k3d-upload-server-image
+
+    - name: Build Everest operator docker image
+      shell: bash
+      run: |
+        make docker-build-operator
+
+    - name: Upload Everest operator image to K3D test cluster
+      shell: bash
+      run: |
+        make k3d-upload-operator-image
+
+    - name: Build Everest CLI binary
+      shell: bash
+      run: |
+        make build-cli-debug
```

Comment:
I am not sure that these steps shall be there. they are not related to K8S at all. Moreover there may be different scenarios what and how exactly to deploy after K8S is ready. So I would split K8S preparation and Everest deployment into separate parts.

Reply:
Okay, in that case, I'm removing this action and copying only those steps here that are needed for the CLI tests.

Please review again